src/db.rs

```diff
@@ -469,6 +478,41 @@ impl Database {
             .await
     }
 
+    /// Runs a database-level command and returns a cursor to the response.
+    pub async fn run_cursor_command(
+        &self,
+        command: Document,
+        options: RunCursorCommandOptions,
+    ) -> Result<Cursor<Document>> {
+        let rcc = RunCommand::new(self.name().to_string(), command, options.selection_criteria.clone(), None)?;
+        let rc_command = RunCursorCommand::new(rcc, options)?; 
+        let client = self.client();
+        client.execute_cursor_operation(rc_command).await
+    }
+
+    /// Runs a database-level command and returns a cursor to the response.
+    pub async fn run_cursor_command_with_session(
+        &self,
+        command: Document,
+        options: impl Into<Option<RunCursorCommandOptions>>,
+        session: &mut ClientSession,
+    ) -> Result<SessionCursor<Document>> {
+        let mut options: Option<RunCursorCommandOptions> = options.into().clone();
+        resolve_selection_criteria_with_session!(self, options, Some(&mut *session))?;
+        let selection_criteria = match options.clone() {
+            Some(options) => options.selection_criteria,
+            None => None,
+        };
```

Comment:
Rather than cloning the entire options struct, it would be more efficient only to clone the `selection_criteria`. You should be able to do this by using a reference to the options, which can be done either like this:

```
let selection_criteria = match options {
    Some(ref options) => options.selection_criteria.clone(),
    None => None,
};
```

or this: 

```
let selection_criteria = match &options {
    Some(options) => options.selection_criteria.clone(),
    None => None,
}
```

(I slightly prefer the first one for no good reason; either is totally fine.)

You can also use `and_then` to achieve the same thing:

```
let selection_criteria = options
    .as_ref()
    .and_then(|options| options.selection_criteria.clone());
```

We use `as_ref` to get a reference to the struct inside the `Option` rather than an owned version of it so that we can continue to use it later. `and_then` does something called a flat map (you can read more about it [here](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then)), which is useful when you want to avoid getting data that's wrapped in multiple layers of `Option`s/`Result`s.

All three of these options would work here, but it's generally considered best practice to use `and_then` rather than an explicit `match` in situations like these.